pkgs/development/python-modules/auto-gpt-plugin-template/default.nix

```
@@ -0,0 +1,35 @@
+{
+  lib,
+  fetchPypi,
+  buildPythonPackage,
+  abstract-singleton,
+  hatchling,
+}:
+
+buildPythonPackage rec {
+  pname = "auto-gpt-plugin-template";
+  version = "0.0.3";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "auto_gpt_plugin_template";
+    hash = "sha256-P1ERD3jbSQqJpRQvzfRBdu8IgSWkGPUHsQDoE+pa/QY=";
+  };
+
+  nativeBuildInputs = [
+    abstract-singleton
+  ];
+
+  propagatedBuildInputs = [
+    hatchling
+  ];
```

```suggestion
  nativeBuildInputs = [
    hatchling
  ];

  propagatedBuildInputs = [
    abstract-singleton
  ];
```
you mixed the inputs up